src/System.Text.Json/tests/Serialization/EnumTests.cs

```diff
@@ -394,5 +394,146 @@ public static void EnumAsSByteMax()
+            json = JsonSerializer.ToString((SampleSByteEnum)0);
+            Assert.Equal("0", json);
+            json = JsonSerializer.ToString((SampleEnumUInt16)0);
+            Assert.Equal("0", json);
+            json = JsonSerializer.ToString((SampleUInt32Enum)0);
+            Assert.Equal("0", json);
+            json = JsonSerializer.ToString((SampleUInt64Enum)0);
+            Assert.Equal("0", json);
+            json = JsonSerializer.ToString((SampleInt64Enum)0);
+            Assert.Equal("0", json);
+        }
+
+        [Fact]
+        public static void ToString_MaxValue_ReturnsMaxValue()
+        {
+            string json = JsonSerializer.ToString((SampleByteEnum)byte.MaxValue);
+            Assert.Equal(byte.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleSByteEnum)sbyte.MaxValue);
+            Assert.Equal(sbyte.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleEnumUInt16)ushort.MaxValue);
+            Assert.Equal(ushort.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt32Enum)UInt32.MaxValue);
+            Assert.Equal(UInt32.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt64Enum)UInt64.MaxValue);
+            Assert.Equal(UInt64.MaxValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleInt64Enum)long.MaxValue);
+            Assert.Equal(long.MaxValue.ToString(), json);
+        }
+
+        [Fact]
+        public static void ToString_MinValue_ReturnsMinValue()
+        {
+            string json = JsonSerializer.ToString((SampleByteEnum)byte.MinValue);
+            Assert.Equal(byte.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleSByteEnum)sbyte.MinValue);
+            Assert.Equal(sbyte.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleEnumUInt16)ushort.MinValue);
+            Assert.Equal(ushort.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt32Enum)UInt32.MinValue);
+            Assert.Equal(UInt32.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleUInt64Enum)UInt64.MinValue);
+            Assert.Equal(UInt64.MinValue.ToString(), json);
+            json = JsonSerializer.ToString((SampleInt64Enum)long.MinValue);
+            Assert.Equal(long.MinValue.ToString(), json);
+        }
+
+        [Theory]
+        [InlineData(byte.MaxValue + 1, (SampleByteEnum)0)]
+        [InlineData(byte.MaxValue + 2, (SampleByteEnum)1)]
+        [InlineData(byte.MaxValue + 13, (SampleByteEnum)12)]
+        [InlineData(byte.MaxValue * 2, (SampleByteEnum)byte.MaxValue - 1)]
+        public static void ToString_Byte(ulong i, SampleByteEnum enumValue)
+        {
+            string json = JsonSerializer.ToString(enumValue);
+            Assert.Equal(((i - 1) % byte.MaxValue).ToString(), json);
+        }
+
+        [InlineData(sbyte.MaxValue + 1, (SampleSByteEnum)0)]
```

Comment:
nit: I know this isn't related to this particular PR, but can you change all the enums so they are consistently named? If we are going with `Sample<Numeric Type>Enum`, let's have all of them follow that format. Currently there's a mix of of formats (e.g. `SampleSByteEnum` vs `SampleEnumUInt16`).

I only mention it since it would help make sure all numeric types are covered and some of the copy/paste typos will be easier to detect.

Reply:
I noticed this as well. but kept unchanged because it was unrelated to this PR. but since it's a small change I can include it in the PR anyways.